systemd/systemd.go

```diff
@@ -405,16 +406,45 @@ func (l Log) String() string {
 	return fmt.Sprintf("%s %s %s", l.Timestamp(), l.SID(), l.Message())
 }
 
+// UseFuse detects if we should be using squashfuse instead
```

Comment:
Ok, so we could remove the squashfs logic from WriteMountUnitFile and just require all callers to pass the right fstype (fuse.squashfs if they need to use fuse), then make UseFuse a bit more generic and move it to osutil. Would that make sense or were you thinking about something else?